### DISCUSSION TITLE
QueryString on v2.6

### DISCUSSION THREAD

**Discussion by Krato** [livewire/livewire#3734] [7. Bugs (archived)] [answered]
Hello everyone.

After upgrade to v2.6 the Livewire components with queryString are not working as expected.

For example, I have a route to a controller that returns a view:

`Route::get('appraisals/{appraisal}/report', [AppraisalController::class, 'report'])->name('report');`

In the view I call a Livewire Component with a queryString var:

```php
protected $queryString = [
    'item' => ['except' => ''],
];
```

When I go to that router clicking in a link, `report` is removed from the URL so the url is:

`https:/mydomain.on/appraisals/12`

Instead

`https:/mydomain.on/appraisals/12/report`

But If I copy & paste the full link into the browser: `https:/mydomain.on/appraisals/12/report` report is not removed from the URL.

Is a problem cause the users usually go to the link navigating. Then if the refresh the page is not reloading the correct page, cause is missing the paramegter. 

I've tested with other components and is happening the same.

I'm missing something from the update? Someone knows why?

Thanks!

**Comment by joshhanley** [comment]
@Krato thanks for the detailed write up! It's much appreciated 🙂

I'm currently trying to replicate the issue. Can you confirm what you mean by this

> When I go to that router clicking in a link

Does that mean you have a link that takes you to the `/appraisals/12/report` url or something else?

In my demo I'm setting up, I have a controller with a view, that has a Livewire component inside of it that uses the query string.

I just now need to confirm what links I should have and where so I can replicate it.

Thanks again!

**Reply by Krato** [reply]
Yes I'ts just link: `<a href="{{ route('appraisal.report', $appraisal->id) }}">Go to reports</a>`

Nothing more.

**Reply by joshhanley** [reply]
@Krato ok thanks!

So if I have this structure, does that sound right:

UsersController -> users list view -> contains an `<a>` link to users page
UserPageController ->user page view -> has Livewire component with query string on it.

Then if I click a link on the users list page to go to a user page, that's when the URL changes?

**Reply by Krato** [reply]
Yes you're right. If you want to see in action, I can send you a private video.

**Reply by joshhanley** [reply]
@Krato yeah if you can that'd be great as I still can't replicate it. If you use discord I'm joshhanley#6022 feel free to send me the video there 🙂

**Reply by Krato** [reply]
Video sent

**Comment by georgeboot** [comment]
We have a similar issue. When I click to a page (full-page component) the URL in the browser address bar changes back to the previous-pages url instantly. Even when I click on from that page, the URL keeps changing back to that first pages url.

This will sometimes even result in a page not loading correctly.

See the following examples: the first time I load the page by clicking a link to it (regular `a`), from the page with incorrect URL in the bar.
The 2nd time, pasted the full URL into a new browser window.

<img width="548" alt="screenshot 22" src="https://user-images.githubusercontent.com/884482/133408159-e7033ff8-fb86-4fac-903b-06f8e795cadd.png">
<img width="552" alt="screenshot 21" src="https://user-images.githubusercontent.com/884482/133408179-0e1624a9-4ae7-4260-ac6d-52748350b383.png">

**Reply by georgeboot** [reply]
Relates to #3735

**Comment by joshhanley** [comment, marked as answer]
I have tagged v2.6.2 which fixes this issue 🙂

Please test it and let us know if it's still happening!

Thanks!

**Reply by georgeboot** [reply]
Confirmed to be working  now. Thanks!

**Reply by Krato** [reply]
Yes. It's working now!

**Reply by joshhanley** [reply]
Awesome! Thanks for testing it out 🙂
